controllers/lib/espejote.libsonnet

```
@@ -1,7 +1,54 @@
 local context = std.extVar('__internal_use_espejote_lib_context');
 local trigger = std.extVar('__internal_use_espejote_lib_trigger');
+local admissionRequest = std.extVar('__internal_use_espejote_lib_admissionrequest');
+
+local admission = {
+  admissionRequest: function() admissionRequest,
+
+  // allowed returns an admission response that allows the operation.
+  allowed: function(msg='') {
+    assert std.isString(msg) : 'msg must be a string, is: %s' % std.manifestJsonMinified(msg),
+    allowed: true,
+    message: msg,
+  },
+
+  // denied returns an admission response that denies the operation.
+  denied: function(msg='') {
+    assert std.isString(msg) : 'msg must be a string, is: %s' % std.manifestJsonMinified(msg),
+    allowed: false,
+    message: msg,
+  },
+
+  // jsonPatchOp returns a JSON patch operation.
+  jsonPatchOp: function(op, path, value=null) {
```

For consistency, I'd add assertions here

```suggestion
  jsonPatchOp: function(op, path, value=null) {
    assert std.isString(op) : 'op must be a string is: %s' % std.manifestJsonMinified(op),
    assert std.isString(path) : 'path must be a string is: %s' % std.manifestJsonMinified(op),
```